File: android/LiveTales/app/src/main/java/com/zed/livetales/models/ConfigModel.java
package com.zed.livetales.models;

import android.util.Log;

import com.kuasars.org.KuasarsError;
import com.kuasars.org.KuasarsListener;
import com.kuasars.org.KuasarsResponse;
import com.kuasars.org.entities.KuasarsEntity;
import com.zed.livetales.ConfigListener;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by jemalpartida on 21/11/2016.
 */

public class ConfigModel
{
    public static void getConfig(final ConfigListener listener)
    {
        KuasarsEntity entity = new KuasarsEntity( "config", false );
        entity.getAsync( "config", "appconfig", new KuasarsListener<KuasarsEntity>() {
            @Override
            public void onComplete(KuasarsResponse<KuasarsEntity> kuasarsResponse)
            {
                KuasarsEntity configEntity = kuasarsResponse.getContent();

                Map<String,String> config = new HashMap<>();

                config.put( "id", configEntity.getId() );
                config.put( "kms_service_url", (String)configEntity.getCustomData().get("kms_service_url") );

                if( listener != null )
                    listener.onReady( config );
            }

            @Override
            public void onError(KuasarsError kuasarsError)
            {
                Log.e( "LiveTales", "Error from Kuasars:" + kuasarsError.getDescription() );
                if( listener != null )
                    listener.onReady( null );
            }
        });
    }
}

File: ios/LiveTales-smartphones/uoat/ConfigModel.swift
//
//  ConfigModel.swift
//  uoat
//
//  Created by Pyro User on 17/11/16.
//  Copyright © 2016 Zed. All rights reserved.
//

import Foundation

class ConfigModel
{
    class func getConfig( onConfigReady:(config:[String:String]?)->Void)
    {
        var config:[String:String]? = nil //[String:String]()
        
        /*
        let criteria:[String:AnyObject] = ["id":"appconfig"]
        let query = KuasarsQuery( criteria, retrievingFields: nil ) as KuasarsQuery
        KuasarsEntity.query( query, entityType: "config", occEnabled: false, completion:{ (response:KuasarsResponse!, error:KuasarsError!) -> Void in
            if( error != nil )
            {
                print( "Error from kuasars: \(error.description)" )
            }
            else
            {
                config = [String:String]()
                let entities = response.contentObjects as! [KuasarsEntity]
                for entity:KuasarsEntity in entities
                {
                    let id:String = entity.ID
                    let kmsServiceUrl:String = (entity.customData!["kms_service_url"])! as! String
                    
                    config!["id"] = id
                    config!["kms_service_url"] = kmsServiceUrl
                }
            }
            onConfigReady( config:config )
        })
        */
        KuasarsEntity.getWithType("config", entityID: "appconfig", occEnabled: false) { (response:KuasarsResponse!, error:KuasarsError!) in
            if( error != nil )
            {
                print( "Error from kuasars: \(error.description)" )
            }
            else
            {
                config = [String:String]()
                let configEntity = response.contentObjects[0] as! KuasarsEntity
                let id:String = configEntity.ID
                let kmsServiceUrl:String = (configEntity.customData!["kms_service_url"])! as! String
                
                config!["id"] = id
                config!["kms_service_url"] = kmsServiceUrl
            }
            onConfigReady( config:config )
        }
    }
}